### PR TITLE
Part1

### PR DESCRIPTION
**Pull request by hilberturi**

### CODE REVIEW

File: Car.h [hunk 1]
```diff
@@ -1,9 +1,17 @@
 #pragma once
 
+#include "Vehicle.h"
+
 struct Car : public Vehicle
```
Comment (matkatmusic) [inline Car.h]:
```
    10) correct any unnecessary keywords related to access specifiers used in the UDTs.
```
- [x] revise

File: Highway.cpp [hunk 2]
```diff
@@ -13,23 +16,45 @@ void Highway::changeSpeed(int newSpeed)
...
+        car->closeWindows();
+    }
+    else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v); motorcycle != nullptr)
+    {
+        motorcycle->lanesplitAndRace(130);
+    }
+    else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+    {
+        semiTruck->headlightFlasher();
+    }
```
Comment (matkatmusic) [inline Highway.cpp]: - use `auto*`.
- you don't need to explicitly check for `nullptr`.  

`nullptr` is effectively `0`, and the computer interprets `0` as `false`.  
if the `dynamic_cast` fails, the pointer will be `nullptr`, which evaluates in this context to `false`

- [x] revise

File: Highway.cpp [hunk 3]
```diff
@@ -13,23 +16,45 @@ void Highway::changeSpeed(int newSpeed)
...
+        car->tryToEvade();
+    }
+    else if (Motorcycle* motorcycle = dynamic_cast<Motorcycle*>(v); motorcycle != nullptr)
+    {
+        motorcycle->tryToEvade();
+    }
+    else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+    {
+        semiTruck->pullOver();
+    }
```
Comment (matkatmusic) [inline Highway.cpp]: - use `auto*`.
- you don't need to explicitly check for `nullptr`.  

`nullptr` is effectively `0`, and the computer interprets `0` as `false`.  
if the `dynamic_cast` fails, the pointer will be `nullptr`, which evaluates in this context to `false`

- [x] revise

File: HighwayPatrol.cpp [hunk 4]
```diff
@@ -28,10 +35,30 @@ void HighwayPatrol::pullOver( Vehicle* v, bool willArrest, Highway* h )
...
+            vehicleType = "MOTORCYCLE";
+        }
+        else if (SemiTruck* semiTruck = dynamic_cast<SemiTruck*>(v); semiTruck != nullptr)
+        {
+            vehicleType = "SEMITRUCK";
+        }
+        else if (HighwayPatrol* highwayPatrol = dynamic_cast<HighwayPatrol*>(v); highwayPatrol != nullptr)
+        {
+            vehicleType = "HIGHWAY PATROL";
+        }
```
Comment (matkatmusic) [inline HighwayPatrol.cpp]: - use `auto*`.
- you don't need to explicitly check for `nullptr`.  

`nullptr` is effectively `0`, and the computer interprets `0` as `false`.  
if the `dynamic_cast` fails, the pointer will be `nullptr`, which evaluates in this context to `false`

- [x] revise

File: SemiTruck.h [hunk 5]
```diff
@@ -1,2 +1,20 @@
 #pragma once
 
+#include "Vehicle.h"
+
+struct SemiTruck : public Vehicle
```
Comment (matkatmusic) [inline SemiTruck.h]:
```
    10) correct any unnecessary keywords related to access specifiers used in the UDTs.
```
- [x] revise

File: Motorcycle.h [hunk 6]
```diff
@@ -1,12 +1,17 @@
 #pragma once
 
-#include <iostream>
+#include "Vehicle.h"
 #include <limits>
 
 struct Motorcycle : public Vehicle
```
Comment (matkatmusic) [inline Motorcycle.h]:
```
    10) correct any unnecessary keywords related to access specifiers used in the UDTs.
```
- [x] revise

**Review by matkatmusic — CHANGES_REQUESTED** [review]

File: Highway.cpp (hunk 2 above)
Comment (hilberturi) [reply Highway.cpp]: Well, the JUCE coding standard forbids using a pointer directly for a boolean test.
That's why I included the explicit check for nullptr.

**Review by matkatmusic — APPROVED** [review]

**Pull request merged**